src/main/scala/is/hail/expr/AST.scala

```diff
@@ -236,7 +236,7 @@ sealed abstract class AST(pos: Position, subexprs: Array[AST] = Array.empty) {
   def runAggregator(ec: EvalContext): CPS[Any] = {
     val typedNames = ec.st.toSeq
       .sortBy { case (_, (i, _)) => i }
-      .map { case (name, (_, typ)) => (name, typ) }
+      .map { case (name, (_, typ)) => (name, typ.scalaClassTag.asInstanceOf[ClassTag[AnyRef]]) }
```

Comment:
These casts are tedious.  Let's see how it looks when you address the above comments.  I think some C[Any] and C[AnyRef] in the code should actually be C[_].